From b627d004f15a6cb088cf2fe31434446328df651f Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Mon, 26 Oct 2026 22:11:17 +0900
Subject: [PATCH 03/15] u-boot: Switch to upstream environment and distro boot

Marek added the PCIe firmware and bootloader update environment and
switched to distro boot in sparrowhawk.h upstream (9e4253d2dc8e,
3511e1e3fbfe, 17baf93381a4). These are included in the current SRCREV,
so drop the local overrides:

- sparrowhawk.env and env_source_file.cfg: duplicated and overrode the
  upstream environment. The autoconf_* scripts are replaced by the
  upstream bootcmd_{mmc0,nvme0,usb0} scripts.
- mmc_boot.cfg: overrode CONFIG_BOOTCOMMAND and CONFIG_BOOTARGS, which
  prevented the upstream distro boot (CONFIG_DISTRO_DEFAULTS) from being
  used. It now searches MMC, NVMe, USB and DHCP for fitImage, boot
  script or extlinux.conf.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 recipes-bsp/u-boot/u-boot/env_source_file.cfg | 1 -
 recipes-bsp/u-boot/u-boot/mmc_boot.cfg        | 2 --
 recipes-bsp/u-boot/u-boot/sparrowhawk.env     | 9 ---------
 recipes-bsp/u-boot/u-boot_git.bb              | 3 ---
 4 files changed, 15 deletions(-)
 delete mode 100644 recipes-bsp/u-boot/u-boot/env_source_file.cfg
 delete mode 100644 recipes-bsp/u-boot/u-boot/mmc_boot.cfg
 delete mode 100644 recipes-bsp/u-boot/u-boot/sparrowhawk.env

diff --git a/recipes-bsp/u-boot/u-boot/env_source_file.cfg b/recipes-bsp/u-boot/u-boot/env_source_file.cfg
deleted file mode 100644
index 5d8f3db..0000000
--- a/recipes-bsp/u-boot/u-boot/env_source_file.cfg
+++ /dev/null
@@ -1 +0,0 @@
-CONFIG_ENV_SOURCE_FILE="sparrowhawk"
diff --git a/recipes-bsp/u-boot/u-boot/mmc_boot.cfg b/recipes-bsp/u-boot/u-boot/mmc_boot.cfg
deleted file mode 100644
index f1a8cbf..0000000
--- a/recipes-bsp/u-boot/u-boot/mmc_boot.cfg
+++ /dev/null
@@ -1,2 +0,0 @@
-CONFIG_BOOTARGS="rw root=/dev/mmcblk0p1 rootwait"
-CONFIG_BOOTCOMMAND="load mmc 0:1 0x58000000 /boot/fitImage && bootm 0x58000000"
diff --git a/recipes-bsp/u-boot/u-boot/sparrowhawk.env b/recipes-bsp/u-boot/u-boot/sparrowhawk.env
deleted file mode 100644
index d9fdc72..0000000
--- a/recipes-bsp/u-boot/u-boot/sparrowhawk.env
+++ /dev/null
@@ -1,9 +0,0 @@
-autoconf_mmc=setenv bootargs "rw root=/dev/mmcblk0p1 rootwait" && load mmc 0:1 ${loadaddr} /boot/fitImage && source ${loadaddr}:script
-autoconf_nvme=pci e && nvme scan && setenv bootargs "rw root=/dev/nvme0n1p1 rootwait" && load nvme 0:1 ${loadaddr} /boot/fitImage && source ${loadaddr}:script
-autoconf_usb=pci e && usb start && setenv bootargs "rw root=/dev/sda1 rootwait" && load usb 0:1 ${loadaddr} /boot/fitImage && source ${loadaddr}:script
-flash_pcie_fw_to_qspi_from_mmc=load mmc 0:1 ${loadaddr} lib/firmware/rcar_gen4_pcie.bin && sf probe; sf update ${loadaddr} 0x300000 ${filesize}
-renesas_rcar_gen4_load_firmware=run set_pcie_firmware_info;sf probe; sf read ${renesas_rcar_gen4_load_firmware_addr} 0x300000 ${renesas_rcar_gen4_load_firmware_size}
-set_pcie_firmware_info=setenv renesas_rcar_gen4_load_firmware_addr 0x54000000 && setenv renesas_rcar_gen4_load_firmware_size 0x8000
-update_loader_from_mmc=load mmc 0:1 ${loadaddr} flash.bin && sf probe && sf update ${loadaddr} 0 ${filesize} && reset
-update_loader_from_usb=pci e; usb start; load usb 0:1 ${loadaddr} flash.bin && sf probe && sf update ${loadaddr} 0 ${filesize} && reset
-update_loader_from_nvme=pci e; nvme scan; load nvme 0:1 ${loadaddr} flash.bin && sf probe && sf update ${loadaddr} 0 ${filesize} && reset
diff --git a/recipes-bsp/u-boot/u-boot_git.bb b/recipes-bsp/u-boot/u-boot_git.bb
index 1d4d743..cf9829e 100644
--- a/recipes-bsp/u-boot/u-boot_git.bb
+++ b/recipes-bsp/u-boot/u-boot_git.bb
@@ -34,10 +34,7 @@ SRC_URI:append = " file://0003-arm64-dts-renesas-r8a779g3-Remove-reserved-memory
 
 SRC_URI:append = "\
     file://nfs_cmd.cfg \
-    file://mmc_boot.cfg \
     file://preboot.cfg \
-    file://env_source_file.cfg \
-    file://sparrowhawk.env;subdir=git/board/renesas/sparrowhawk/ \
 "
 
 do_deploy:append() {
